### PR TITLE
User friendly forum links

### PR DESCRIPTION
**Pull request by motechFR**
PR for user friendly forum links and some extra cleanup

**1. Forum links navigation format**

All categories:
/forum/

Category view:
/forum/category/<categoryName>

Category view with popup (we could save this as a fast follower):
/forum/category/<categoryName>?postId=<postId>

Single post view*:
/forum/post/<post_path>

**2. Forum links generation**
- Autogenerated by our library; supports japanese characters

**3. Post Summary**
- Now the time icon only shows time post was created at for a better UX

**4. New post from category**
- New post from current category will default to that category
- On "All categories" it defaults to the space default category

### CODE REVIEW

**Review by Devorein — APPROVED** [review]

File: components/forum/components/CategoryMenu.tsx [hunk 1]
```diff
@@ -195,9 +200,7 @@ export function CategoryMenu() {
           }}
         />
         <Button
-          disabled={
-            forumCategoryName.length === 0 || categories.find((category) => category.name === forumCategoryName)
-          }
+          disabled={categories.find((category) => category.name === newForumCategoryName)}
```
Comment (Devorein) [inline components/forum/components/CategoryMenu.tsx:203]: We should disable empty categories from being created

**Review by Devorein — APPROVED** [review]
Looks great. Maybe if a user is in a specific category and deletes that category we could update the url to remove the category as the category itself is switched to `All categories`

**Review by mattcasey — CHANGES_REQUESTED** [review]
Looks good, but one thing: we should sanitize category names so that the path is all lowercase, spaces are replaced with hyphens, and special characters stripped out. For example "category / subcategory" as category results in a 404.

**Comment by Devorein** [comment]
It will be quite difficult to serialize and de-serialize them. Maybe this [npm package](https://www.npmjs.com/package/friendly-url) can help.

**Pull request merged**